src/Morphir/IR/Repo.elm

```diff
@@ -124,9 +126,91 @@ parseNewElmModules fileChanges =
+             let
+                  -- extract IR moduleName from ParsedModule
+                  fromModuleNameParam = parsedModule |> ParsedModule.moduleName |> toIRModuleName repo.packageName
+
+                  -- extract parsedMod Dependencies + format each moduleName to type ModuleName
+                  moduleDependencies: List ModuleName
+                  moduleDependencies =
+                      ParsedModule.importedModules parsedModule
+                           |> List.map (toIRModuleName repo.packageName)
+
+                  -- insert function to DAG Graph
+
+              in
+              moduleDependencies
+                   |> List.foldl (foldLeftInnerFunction)
+       in
+       parsedModules
+           |> List.foldl (foldLeftInitialFunction) finalGraph
+           |> Result.andThen (\functionAResult ->
+               case functionAResult of
+                   Err err ->
+                       case  err of
+                           _ ->
+                               CycleDetected
+                   Ok _ ->
+                       parsedModules
+           )
+    -}
+    let
+        moduleGraph : DAG ModuleName
+        moduleGraph =
+            DAG.empty
+
+        foldFunction : ParsedModule -> Result Error (DAG ModuleName) -> Result Error (DAG ModuleName)
+        foldFunction parsedModule graph =
+            let
+                moduleName : ModuleName
+                moduleName =
+                    ParsedModule.moduleName parsedModule
+                        |> toIRModuleName repo.packageName
+
+                moduleDependencies : List ModuleName
+                moduleDependencies =
+                    ParsedModule.importedModules parsedModule
+                        |> List.map (\modName -> toIRModuleName repo.packageName modName)
+
+                insertEdge : ModuleName -> Result Error (DAG ModuleName) -> Result Error (DAG ModuleName)
+                insertEdge toModule dag =
+                    dag
+                        |> Result.andThen
+                            (\graphValue ->
+                                graphValue
+                                    |> DAG.insertEdge moduleName toModule
+                                    |> Result.mapError
+                                        (\err ->
+                                            case err of
+                                                _ ->
+                                                    CycleDetected moduleName toModule
+                                        )
+                            )
+            in
+            moduleDependencies
+                |> List.foldl insertEdge graph
+    in
+    parsedModules
+        |> List.foldl foldFunction (Ok moduleGraph)
+        |> Result.andThen
+            (\finalGraph ->
+                finalGraph
+                    |> DAG.forwardTopologicalOrdering
+                    |> List.concat
+                    |> List.map2
```

Comment:
You can remove `map2` and change `Result.andThen` to `Result.map`

Reply:
Fixed